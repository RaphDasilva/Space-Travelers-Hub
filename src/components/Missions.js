import React from 'react';
import DisplayMission from './DisplayMission';
import styles from '../styles/Missions.module.css';

function Missions() {
  return (
    <div className={styles.missionsContainer}>
      <div className={styles.missionsHeader}>
        <div>
          <p>Mission</p>
        </div>
        <div>
          <p>Description</p>
        </div>
        <div>
          <p>Status</p>
        </div>
        <div />
      </div>
      <DisplayMission />
    </div>
  );
}

export default Missions;
