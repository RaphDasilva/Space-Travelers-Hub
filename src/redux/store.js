import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketsReducer from './rockets/rockets';

const store = configureStore({
  reducer: {
    mission: missionReducer, rocketsReducer,
  },
});

export default store;
