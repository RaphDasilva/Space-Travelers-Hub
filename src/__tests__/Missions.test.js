import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Missions from '../components/Missions';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Missions component', () => {
  it('renders the Missions component', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(getByText('Mission')).toBeInTheDocument();
    expect(getByText('Description')).toBeInTheDocument();
    expect(getByText('Status')).toBeInTheDocument();
  });
});

describe('Missions component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
