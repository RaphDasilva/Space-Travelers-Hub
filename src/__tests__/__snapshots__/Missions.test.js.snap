// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component should render correctly 1`] = `
<div>
  <div
    class="missionsContainer"
  >
    <div
      class="missionsHeader"
    >
      <div>
        <p>
          Mission
        </p>
      </div>
      <div>
        <p>
          Description
        </p>
      </div>
      <div>
        <p>
          Status
        </p>
      </div>
      <div />
    </div>
    <div />
  </div>
</div>
`;
